modules/preview-react/segmented-control/lib/hooks/useSegmentedControlModel.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import {createModelHook} from '@workday/canvas-kit-react/common';
+import {
+  defaultGetId,
+  useListModel,
+  useOverflowListModel,
+} from '@workday/canvas-kit-react/collection';
+
+export const useSegmentedControlModel = createModelHook({
+  defaultConfig: {
+    ...useOverflowListModel.defaultConfig,
+    /**
+     * Optional id for the whole `SegmentedControl` group. If not provided, a unique id will be created.
+     * @default useUniqueId()
+     */
+    id: '',
+    /**
+     * An initially selected value. This value must match the `data-id` of the `SegmentedControl.Item` component.
+     * If not provided, the first value will be selected.
+     */
+    initialValue: '',
+    /**
+     * Sets disabled state to all segmented control buttons
+     * @default false
+     */
+    disabled: false,
+    /**
+     * Sets size of segmented control container and buttons. Can be `small`, `medium` or `large`.
```

Comment:
```suggestion
     * Sets the size of the segmented control container and its buttons. Can be `small`, `medium` or `large`.
```

Reply:
I was trying to figure out a way to define this to better docs. Thoughts on using @param from [jsdocs](https://jsdoc.app/tags-param.html) for each valid value?

Reply:
I guess @param is more for function description, isn't it?

Reply:
I don't use `@param` in JSDoc and instead add JSDoc to each parameter. At the moment, we don't have a way to extract documentation out of non-props for components, but I plan to add the ability to create documentation objects out of anything to allow us to more easily document utility functions and models.